proto/minder/v1/minder.proto

```diff
@@ -2098,6 +2110,22 @@ message ListProviderClassesResponse {
     repeated string provider_classes = 1;
 }
 
+message ProviderPatch {
+    // to allow generic configuration changes.
+    google.protobuf.Struct config = 1;
+}
+
+message PatchProviderRequest {
+    Context context = 1;
+    ProviderPatch patch = 2;
+
+    google.protobuf.FieldMask update_mask = 3;
```

Comment:
I don't recall exactly how good the library support is for `update_mask`.  It's real handy as a client, but make sure it won't get you into too much trouble in the implementation.

Reply:
I actually don't use it now in the implementation. We might want to reconsider later, but yeah, it seems like a lot of trouble for very questionable gain (Unless the FE devs disgree, going to ping them now for API review)